cmd/podman/images/scp.go

```diff
@@ -0,0 +1,383 @@
+		scpOpts.Load.Quiet = true
+	}
+	scpOpts.ToFrom = true                 // default to loading to the ssh client
+	f, err := ioutil.TempFile("", "temp") // open temp dir for load/save output
+	if err != nil {
+		return err
+	}
+	scpOpts.Save.Output = f.Name()
+	scpOpts.Load.Input = scpOpts.Save.Output
+	if err := parse.ValidateFileName(saveOpts.Output); err != nil {
+		return err
+	}
+	confR, err := config.NewConfig("") // create a hand made config for the remote engine since we might use remote and native at once
+	if err != nil {
+		return errors.Wrapf(err, "could not make config")
+	}
+	abiEng, err := registry.NewImageEngine(cmd, args) // abi native engine
+	if err != nil {
+		return errors.Wrapf(err, "could not make abi engine")
+	}
+	url := ""
+	iden := ""
+	serv := map[string]config.Destination{}
+	cfg, err := config.ReadCustomConfig() // get ready to set ssh destination if necessary
+	if err != nil {
+		return err
+	}
+	conn := ""
+	if strings.Contains(args[0], "::") || len(args) == 2 { // if we have a connection name specified
+		if len(args) == 2 && !(strings.Contains(args[1], "::") && strings.Contains(args[0], "::")) { // if an image is specified, this mean we are loading from our client
+			conn = args[1]
+		} else { // else we are loading from the ssh/remote client
+			scpOpts.ToFrom = false
+			conn = args[0]
+		}
+		splitEnv := strings.SplitN(conn, "::", 2)
+		if len(splitEnv) == 2 {
+			conn = splitEnv[0]
+			if len(args) == 2 && !(strings.Contains(args[1], "::") && strings.Contains(args[0], "::")) && splitEnv[1] != "" { // if we have two arguments and its not remote -> remote
+				scpOpts.Load.Input = splitEnv[1]
+			} else {
+				scpOpts.ImageName = splitEnv[1]
+			}
+		}
+		if len(cfg.Engine.ServiceDestinations) == 0 { // no connections in list >:/
+			logrus.Warnf("Unknown connection name given. Please use system connection add to specify the default remote socket location")
+		}
+
+		for name, data := range cfg.Engine.ServiceDestinations { // loop through all of the connections until we find outs
+			if name == conn {
+				iden = data.Identity
+				url = data.URI
+			}
+		}
+		if iden == "" && url == "" { // no match, warn user and do a manual connection.
+			url = "ssh://" + conn
+			logrus.Warnf("Unknown connection name given. Please use system connection add to specify the default remote socket location")
+		}
+	} else { // else we default to remote or whatever the default connection is
+		connStr := cfg.Engine.ActiveService
+		if len(cfg.Engine.ServiceDestinations) == 0 {
+			logrus.Warnf("Unknown connection name given. Please use system connection add to specify the default remote socket location")
+		}
+
+		for name, data := range cfg.Engine.ServiceDestinations {
+			if name == connStr {
+				iden = data.Identity
+				url = data.URI
+			}
+		}
+		dst := config.Destination{}
+		if iden != "" {
+			dst.URI = url
+			dst.Identity = iden
+		} else {
+			dst.URI = url
+		}
```

Comment:
This is exists in very similar form twice, and looks like a good candidate for a separate helper function.

Reply:
Still outstanding, I think. It’s surprising that the two look so different — OTOH maybe they are different for a good reason, and and should not be consolidated. Which one is it?

Reply:
Only one instance now.

Reply:
I think this is outdated as I cannot find this in the most recent push.

Reply:
Yup, looks good.